src/common/components/signUpModal/signUpModal.component.js

```diff
@@ -1,81 +1,326 @@
 import angular from 'angular'
-import 'angular-gettext'
-import 'angular-messages'
-import sessionService from 'common/services/session/session.service'
-import showErrors from 'common/filters/showErrors.filter'
+import includes from 'lodash/includes'
+import assign from 'lodash/assign'
+import pick from 'lodash/pick'
+import 'rxjs/add/operator/finally'
+import { Observable } from 'rxjs/Observable'
+import OktaSignIn from '@okta/okta-signin-widget'
+import sessionService, { Roles } from 'common/services/session/session.service'
+import orderService from 'common/services/api/order.service'
 import template from './signUpModal.tpl.html'
-import valueMatch from 'common/directives/valueMatch.directive'
+import cartService from 'common/services/api/cart.service'
+require('assets/okta-sign-in/css/okta-sign-in.min.css')
 
 const componentName = 'signUpModal'
 
 class SignUpModalController {
   /* @ngInject */
-  constructor (gettext, sessionService) {
-    this.gettext = gettext
+  constructor ($log, $scope, $location, $translate, sessionService, cartService, orderService, envService) {
+    this.$log = $log
+    this.$scope = $scope
+    this.$location = $location
+    this.$translate = $translate
     this.sessionService = sessionService
+    this.orderService = orderService
+    this.cartService = cartService
+    this.imgDomain = envService.read('imgDomain')
+    this.publicCru = envService.read('publicCru')
   }
 
   $onInit () {
-    this.modalTitle = this.gettext('Sign Up')
-    this.isLoading = false
-    this.setPristine()
-  }
+    if (includes([Roles.identified, Roles.registered], this.sessionService.getRole())) {
+      this.onStateChange({ state: 'sign-in' })
+    }
+    this.currentStep = 1
+    this.donorDetails = {}
+    this.loadDonorDetails().finally(() => {
+      this.setUpSignUpWidget()
+    }).subscribe()
```

Comment:
```suggestion
    this.loadDonorDetails().subscribe({
      finally: () => this.setUpSignUpWidget()
    })
```

Reply:
I don't think RxJS 5 supports using `finally ` inside `subscribe()`.

Are you wanting to do something like this? If so, `complete` only runs when the Observable successfully completes, so if it errors it will not fire.
```
this.loadDonorDetails().subscribe({
  complete: () => this.setUpSignUpWidget(),
});
```

Reply:
I'm sorry, you're right. I was thinking of `complete`, but you're right that it doesn't fire when the observable errors.